File: src/main/java/ru/ifmo/is/mfl/movieviews/dto/MovieViewWithoutMovieDto.java
package ru.ifmo.is.mfl.movieviews.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;
import ru.ifmo.is.mfl.common.framework.dto.CrudDto;
import ru.ifmo.is.mfl.users.dto.UserDto;

import java.time.Instant;

@Data
@EqualsAndHashCode(callSuper = true)
public class MovieViewWithoutMovieDto extends CrudDto {
  private int id;
  private UserDto user;
  private Instant watchDate;
}

File: frontend/src/interfaces/movieview/dto/MovieViewWithoutMovieDto.ts
import { CrudDto } from "@/interfaces/crud/dto/CrudDto";
import { UserDto } from "@/interfaces/user/dto/UserDto";
import { LocalDate } from "@/types/LocalDate";

export interface MovieViewWithoutMovieDto extends CrudDto {
  user: UserDto;
  watchDate: LocalDate;
}